pinot-core/src/main/java/org/apache/pinot/core/operator/query/DictionaryBasedDistinctOperator.java

```diff
@@ -134,6 +128,20 @@ private DistinctTable buildResult() {
     return new DistinctTable(dataSchema, records, _nullHandlingEnabled);
   }
 
+  private void iterateOnDictionary(int dictLength, int actualLimit, List<Record> records, boolean isAscending) {
+    if (isAscending) {
```

Comment:
(nit) I suggest separating this method into 2  to avoid doing the `if (_isAscending)` branch twice 

cc @jasperjiaguo

Reply:
Done